### PR TITLE
fix: (scala3) insert correct position for implement-all code action on class with type parameters

### PR DESCRIPTION
**Pull request by tanishiking**
fix https://github.com/scalameta/metals/issues/4062

Implement-all code action for Scala3 infers the insert position based on the existing members if there are no braces.

### fix1
Previously, Metals infer the insert position based on the class's constructor parameters and type parameters.
For example,

```scala
class Concrete[T] extends Base
```

Metals had been inferring the position based on `T` and we end up
inserting stub implementation in the wrong positions.

This commit excludes type params from the "existing declaration".

### fix2
Extension methods shouldn't have an `override` keyword. For example, the following code will be invalid.

```scala
class Concrete extends Base:
  override extension (x: Int) def foo: Int = ???
```

This commit excludes the `override` keyword if the completing method is an extension method.
It's ideal for adding `override` before `def` like 

```scala
class Concrete extends Base:
  extension (x: Int) override def foo: Int = ???
```

But it's a bit tricky to insert the keyword there because the `extension (x: Int) def ...` part is generated by MetalsPrinter.

---

The code action in Scala3 still has a problem with higher kind types, for example

```scala
trait NodeDb[F[_]]:
  type N

  extension (node: N)
    def parent: F[Option[N]]

private final class InMemoryNodeDb[F[_]] extends NodeDb[F]
```

will be 

```scala
trait NodeDb[F[_]]:
  type N

  extension (node: N)
    def parent: F[Option[N]]

private final class InMemoryNodeDb[F[_]] extends NodeDb[F] {
  extension (node: NodeDb.this.N) def parent: Any = ???
}
```

that doesn't compile. But let's work on this issue in another PR :)

### CODE REVIEW

**Comment by tanishiking** [comment]
Also, it might be ideal to complete like 


```scala
class Concrete extends Base:
  extension (x: Int)
    def foo: Int = ???
    def bar: Int = ???
```

instead of 

```scala
class Concrete extends Base:
  extension (x: Int) def foo: Int = ???
  extension (x: Int) def bar: Int = ???
```

but this PR doesn't do that because

- It requires complete re-implementation of a building text part of the code-action
- Not sure there's a demand

Let's do this if someone raise an issue in another PR

File: tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala [hunk 1]
```diff
@@ -935,6 +965,62 @@ class AutoImplementAbstractMembersSuite extends BaseCodeActionSuite {
...
+    """|package a
+       |
+       |trait Base:
+       |  extension (x: Int)
+       |    def foo: Int
+       |    def bar: String
+       |
+       |class Concrete extends Base {
+       |
+       |  exntesion (x: T) def foo: Int = ???
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala:988]:
```suggestion
       |  extension (x: Int) def foo: Int = ???
```

File: tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala [hunk 2]
```diff
@@ -935,6 +965,62 @@ class AutoImplementAbstractMembersSuite extends BaseCodeActionSuite {
...
+       |
+       |  extension (x: T) def bar: Int = ???
```
Comment (tgodzik) [inline tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala:990]:
```suggestion
       |  extension (x: Int) def bar: Int = ???
```

Comment (tanishiking) [reply tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala]: Auch, fixed! https://github.com/scalameta/metals/pull/4081/commits/724f72652552ddec7ab5d1b62568ba13df774623

**Comment by tanishiking** [comment]
> Maybe we should make sure that MetalsPrinter inserts that correctly?

Added `override` keyword for extension methods

https://github.com/scalameta/metals/pull/4081/commits/724f72652552ddec7ab5d1b62568ba13df774623

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala [hunk 3]
```diff
@@ -353,18 +358,14 @@ object OverrideCompletions:
...
 
-    val asciOverrideDef: String =
-      if sym.is(Abstract) then ""
-      else overrideKeyword
-
-    val overrideDef: String = config.overrideDefFormat() match
+    val overrideDefLabel: String = config.overrideDefFormat() match
       case OverrideDefFormat.Unicode =>
-        if sym.is(Abstract) then "🔼 "
+        if sym.is(Deferred) then "🔼 "
```
Comment (tanishiking) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala:366]: This is irrelevant to the main change, but this change makes `OverrideConfigSuite`'s `unitcode` test closer to Scala2.

**Pull request merged**